SocketRocket/SRWebSocket.m

```diff
@@ -1113,6 +1113,8 @@ - (void)_pumpWriting
             }
         }
 
+        self.readyState = SR_CLOSED;
```

Comment:
Shouldn't this code go inside the `if(!_failed)`?
What's the state of the WebSocket if we send a command to close it and it fails to close for whatever reason? Is it in closing state? is it open?

Reply:
@cipolleschi Thank you! You're right, it does make more sense to put it inside `if(!_failed)`. I fixed it. 

https://github.com/facebookincubator/SocketRocket/pull/693/commits/2d81c50baf61b2ebbfa7a24c3c050abb3dde2092